### PR TITLE
Fix zoom limits

### PR DESCRIPTION
**Pull request by theplatapi**
This fixes issue #3549. The camera's `flyTo` method now checks the `screenSpaceCameraController`'s min and max zoom settings and adjusts the final destination height accordingly.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Thanks @theplatapi!  Can you also update CHANGES.md in the root directory?

@bagnell can you please review this?  We have a CLA from @theplatapi.

**Comment by theplatapi** [comment]
I updated the CHANGES.md file.

**Comment by theplatapi** [comment]
I fixed the merge conflicts with master. Let me know if you need anything else for this PR.

**Comment by pjcozzi** [comment]
@bagnell this is ready for review.

File: Source/Scene/Camera.js [hunk 1]
```diff
@@ -2552,6 +2552,23 @@ define([
             destination = this.getRectangleCameraCoordinates(destination, scratchFlyToDestination);
         }
 
+        var sscc = this._scene.screenSpaceCameraController;
+        var ellipsoid = this._scene.globe.ellipsoid;
```
Comment (bagnell) [inline Source/Scene/Camera.js]: In practice, `this._scene.globe` should not be `undefined`, but it may be for testing or debugging. Use `this._scene.mapProjection.ellipsoid`.

File: Source/Scene/Camera.js [hunk 2]
```diff
@@ -2552,6 +2552,23 @@ define([
...
+        var destinationCartographic = Cartographic.fromCartesian(destination);
+
+        // Make sure camera doesn't zoom outside set limits
+        if (defined(sscc.minimumZoomDistance) && defined(sscc.maximumZoomDistance)) {
+            destinationCartographic.height = CesiumMath.clamp(destinationCartographic.height, sscc.minimumZoomDistance, sscc.maximumZoomDistance);
```
Comment (bagnell) [inline Source/Scene/Camera.js]: You only need to check that `sscc` is defined. If it is, `minimumZoomDistance` and `maximumZoomDistance` are assumed to be defined.

**Comment by bagnell** [comment]
@theplatapi I just have those two comments. Also could you merge in master again? Thanks!

**Comment by hpinkos** [comment]
@bagnell Is there any easy way to add a test for this?

**Comment by bagnell** [comment]
@hpinkos Yes! Thanks, for mentioning it. @theplatapi Could you add tests for flights that would go above and below the minimum heights and expect it to be clamped.

**Comment by theplatapi** [comment]
I addressed the comments and added tests.

**Comment by theplatapi** [comment]
I fixed the new conflicts with master. Do the changes look good?

**Comment by bagnell** [comment]
Looks good. Thanks, @theplatapi!

**Comment by burn123** [comment]
This appears to fail on custom geocoders. I am using the following code

    OpenStreetMapNominatimGeocoder.prototype.geocode = function (input) {
    var endpoint = 'http://nominatim.openstreetmap.org/search?';
    var query = 'format=json&q=' + input;
    var requestString = endpoint + query;
    return Cesium.loadJson(requestString)
        .then(function (results) {
            var bboxDegrees;
            return results.map(function (resultObject) {
                bboxDegrees = resultObject.boundingbox;
                return {
                    displayName: resultObject.display_name,
                    destination: Cesium.Rectangle.fromDegrees(
                        bboxDegrees[2],
                        bboxDegrees[0],
                        bboxDegrees[3],
                        bboxDegrees[1]
                    )
                };
            });
        });
    };

and this zooms in past the defined max zoom

**Comment by pjcozzi** [comment]
@burn123 could you please submit a new issue and link to this?  Otherwise, I don't know if everyone will see your comment on this closed pull request.  Thanks!
